assets/RNAseq_report.Rmd

```diff
@@ -1780,6 +1785,180 @@ pg[['x']][['layout']][['annotations']][[1]][['y']] <- -0.05
 ```
 
+
+```{r enhanced_volcano, echo=FALSE, message=FALSE, warning=FALSE}
+
+library(EnhancedVolcano)
+library(ggplot2)
+
+theme_set(theme_bw(base_family = "ArialMT") +
+theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(family="ArialMT")))
+
+
+thousandFormat <- function(x){
+  formatC(x, format='f', big.mark=',', digits=0)
+  }
+
+allgenes_files <- list.files(path=paste0(wd, "/differential_gene_expression/allgenes/"))
+volcano_list <-  c()
+contrast_list <- c()
+
+
+for (file in allgenes_files){
+
+    # Reading DE genes list
+    fname <- tools::file_path_sans_ext(basename(file))
+    DE_genes <- read.csv(file = paste0(wd, "/differential_gene_expression/allgenes/", file), sep="\t", header = T)
+    DE_genes$contrast <- rep(fname, nrow(DE_genes))
+    contrast_len <- sapply(strsplit(fname, '_'), length)
+    # Rename Contrast for plot
+    # DE_contrast_condition_conditionName_conditionA_vs_conditionB --> conditionA vs conditionB
+    contrast_list <- c(contrast_list, sapply(strsplit(fname, '_'), function(x) paste(x[5:contrast_len], collapse=' ')))
+
+    # label: gene_name
+    # x: log2FoldChange
+    # y: -log10(padj)
+
+    DE_genes <- na.omit(DE_genes)
+
+    # convert zeros to small value (as in EnhancedVolcano)
+    DE_genes[which(DE_genes[['padj']] == 0), 'padj'] <- min(
+    DE_genes[which(DE_genes[['padj']] != 0), 'padj'], na.rm = TRUE) * 10^-1
+
+    # Get Number of DE features
+    numDE <- DE_genes %>%
+        filter(padj <= params$adj_pval_threshold, abs(log2FoldChange) >= params$logFC_threshold) %>%
+        nrow()
+
+    # Plot old volcano version static and save
+    DE_genes$logpval <- -log10(DE_genes$padj)
+    # Save true value for hovertext
+    DE_genes$truelogpval <- DE_genes$logpval
+    DE_genes <- DE_genes %>% mutate(color = ifelse((abs(DE_genes$log2FoldChange) >= params$logFC_threshold) &
+                        (DE_genes$logpval >= -log10(params$adj_pval_threshold)),
+                        "red", "black"))
+    log2FoldChange_min <- min(DE_genes$log2FoldChange)
+    log2FoldChange_max <- max(DE_genes$log2FoldChange)
+    log2FoldChange_quotient <- length(seq(log2FoldChange_min, log2FoldChange_max, 10))
+
+    pg <- ggplot(DE_genes, aes(x=log2FoldChange, y=logpval, text=paste("Gene: ", gene_name,
+                                                                        "<br>", "Log2FC: ", formatC(log2FoldChange, digits=2),
+                                                                        "<br>", "-Log10padj: ", formatC(truelogpval, digits=2, format='f')))) +
+            geom_point(alpha=0.5, aes(colour=ifelse(color == "red", "DE features",
+                                                                    "Non_DE features")),
+                                                                    na.rm=TRUE) +
```

Comment:
The ident is not intuitive to me, just a comment

Reply:
That is true! I will change the format here

Reply:
Should be fixed now